M/MLX/build_tarballs.jl

```diff
@@ -25,16 +25,19 @@ fi
 
 cd $WORKSPACE/srcdir/mlx
 
-atomic_patch -p1 ../patches/cmake_system_processor-arm64-aarch64.patch
-atomic_patch -p1 ../patches/cmake-x86_64-apple-darwin.patch
+# Apply patches merged upstream after v0.21.0 was tagged
+git config user.name "John Doe"
+git config user.email john@doe.com
+git remote add upstream https://github.com/ml-explore/mlx.git
+git fetch upstream
+git checkout -B patches
+git cherry-pick 698e63a608bbee43bd70a76602dbf2bd6636877f # CMake: Build with dlfcn-win32 to have dlopen etc. on win32 (#1628)
```

Comment:
I haven't seen this pattern before – using the build script to download more code. Usually the recipes put patches into a `bundled/patches` directory.

Reply:
Nope, it’s new :-)

It’s just that these patches were already upstreamed, so it seems cleaner to just get them from the source, but we don’t want a newer commit as other (possibly breaking) changes has also been merged.

Reply:
Why not adding a patch (which are also cached, contrary to what you're doing here)?  That looks much cleaner than doing all this needless dance.

Reply:
I can re-add the patches, though they will be gone by v0.21.1 (just released).

… they’re should be in this branch already - but not at the latest commit.

I would still like to build v0.21.0 as this is the version used by MLX C v0.1.0.

Reply:
> I can re-add the patches, though they will be gone by v0.21.1 (just released).

That's just normal: we backport patches all the time.  Fiddling with the git repo looks more messy to me.

Reply:
I beg to differ, but I've restored the patches now :-)